File: android/app/src/main/java/com/noop/brand/CustomerFacingBrand.kt
package com.noop.brand

/**
 * Removes transport-vendor wording from dynamic customer-facing text.
 *
 * Wire identifiers, persisted source IDs, and provenance records remain unchanged. This covers
 * runtime strings such as historical release notes and the user-visible strap log, which cannot rely
 * on Android string-resource overrides.
 */
internal object CustomerFacingBrand {
    private val replacements = listOf(
        Regex("""\bmy-whoop-noop\b""", RegexOption.IGNORE_CASE) to "primary-band-on-device",
        Regex("""\bmy-whoop\b""", RegexOption.IGNORE_CASE) to "primary-band",
        Regex("""\bopenwhoop\b""", RegexOption.IGNORE_CASE) to "NOOP legacy storage",
        Regex("""\bwhoopstore\b""", RegexOption.IGNORE_CASE) to "local store",
        Regex("""\bwhoopimporter\b""", RegexOption.IGNORE_CASE) to "wearable importer",
        Regex("""\bwhoop_live_hr_in_adv_ind_pkt\b""", RegexOption.IGNORE_CASE) to "band broadcast setting",
        Regex("""\bwhoop[- ]compatible\b""", RegexOption.IGNORE_CASE) to "wearable-compatible",
        Regex("""\bwhoop(?=\d\w*[-_])""", RegexOption.IGNORE_CASE) to "band-",
        Regex("""\bwhoop\s*5(?:\.0)?(?:\s*/?\s*mg)?\b""", RegexOption.IGNORE_CASE) to "newer band",
        Regex("""\bwhoop\s*4(?:\.0)?\b""", RegexOption.IGNORE_CASE) to "legacy band",
        Regex("""\bwhoop\s+mg\b""", RegexOption.IGNORE_CASE) to "ECG-capable band",
        Regex("""\bwhoop\s+exports\b""", RegexOption.IGNORE_CASE) to "wearable exports",
        Regex("""\bwhoop\s+export\b""", RegexOption.IGNORE_CASE) to "wearable export",
        Regex("""\bwhoop\s+imports\b""", RegexOption.IGNORE_CASE) to "wearable imports",
        Regex("""\bwhoop\s+import\b""", RegexOption.IGNORE_CASE) to "wearable import",
        Regex("""\bwhoop[-_ ]csv\b""", RegexOption.IGNORE_CASE) to "wearable-csv",
        Regex("""\bwhoop\s+app\b""", RegexOption.IGNORE_CASE) to "band app",
        Regex("""\bwhoop-style\b""", RegexOption.IGNORE_CASE) to "wearable-style",
        Regex("""\bwhoop(?:'s|’s)\b""", RegexOption.IGNORE_CASE) to "the provider's",
        Regex("""\bwhoop(?=\d)""", RegexOption.IGNORE_CASE) to "band-",
        Regex("""whoop""", RegexOption.IGNORE_CASE) to "compatible band",
    )

    fun text(raw: String): String =
        replacements.fold(raw) { value, (pattern, replacement) ->
            value.replace(pattern, replacement)
        }
}

File: Strand/System/CustomerFacingBrand.swift
import Foundation

/// Removes transport-vendor wording from dynamic text before it reaches customer UI.
///
/// Protocol identifiers, persisted source IDs, and provenance records stay unchanged. This is for
/// strings such as historical release notes and diagnostic lines that cannot use String Catalog
/// overrides because they are rendered from runtime `String` values.
enum CustomerFacingBrand {
    private static let replacements: [(pattern: String, replacement: String)] = [
        (#"(?i)\bmy-whoop-noop\b"#, "primary-band-on-device"),
        (#"(?i)\bmy-whoop\b"#, "primary-band"),
        (#"(?i)\bopenwhoop\b"#, "NOOP legacy storage"),
        (#"(?i)\bwhoopstore\b"#, "local store"),
        (#"(?i)\bwhoopimporter\b"#, "wearable importer"),
        (#"(?i)\bwhoop_live_hr_in_adv_ind_pkt\b"#, "band broadcast setting"),
        (#"(?i)\bwhoop[- ]compatible\b"#, "wearable-compatible"),
        (#"(?i)\bwhoop(?=\d\w*[-_])"#, "band-"),
        (#"(?i)\bwhoop\s*5(?:\.0)?(?:\s*/?\s*mg)?\b"#, "newer band"),
        (#"(?i)\bwhoop\s*4(?:\.0)?\b"#, "legacy band"),
        (#"(?i)\bwhoop\s+mg\b"#, "ECG-capable band"),
        (#"(?i)\bwhoop\s+exports\b"#, "wearable exports"),
        (#"(?i)\bwhoop\s+export\b"#, "wearable export"),
        (#"(?i)\bwhoop\s+imports\b"#, "wearable imports"),
        (#"(?i)\bwhoop\s+import\b"#, "wearable import"),
        (#"(?i)\bwhoop[-_ ]csv\b"#, "wearable-csv"),
        (#"(?i)\bwhoop\s+app\b"#, "band app"),
        (#"(?i)\bwhoop-style\b"#, "wearable-style"),
        (#"(?i)\bwhoop(?:'s|’s)\b"#, "the provider's"),
        (#"(?i)\bwhoop(?=\d)"#, "band-"),
        (#"(?i)whoop"#, "compatible band"),
    ]

    static func text(_ raw: String) -> String {
        replacements.reduce(raw) { value, item in
            value.replacingOccurrences(
                of: item.pattern,
                with: item.replacement,
                options: .regularExpression
            )
        }
    }
}
